_posts/2024-04-18-ship-cars-leverages-quarkus-to-reach-its-goals.adoc

```diff
@@ -0,0 +1,146 @@
+
+4. **Dev UI**: Developers can configure extensions, monitor the application, and test components with great ease.
+
+5. **Unified config**: All of the application's configurations are consolidated in one place, improving accessibility.
+
+6. Standards-based
+
+== Embracing Quarkus extensions
+
+Quarkus Extensions are pre-configured feature sets designed to simplify several common tasks during application development.
+They offer an efficient way to imbibe new capabilities or direct integrations in your project with minimum effort.
+
+In our organization, we managed to implement our internal extensions swiftly, effectively addressing maintenance issues and configuration incompatibilities we encountered earlier while trying to create native images.
+Today, we benefit from an extension hub that quells all previous concerns and enhances our productivity.
+
+While Quarkus extensions are powerful tools offering deep integration, optimization, and enhanced developer experience, it's essential to weigh the trade-offs and consider if simpler solutions like standard JAR libraries might suit the need better.
+
+== Looking ahead
+
+In the graphical representation below, I want to illustrate the inherent relationship between the process of adopting Quarkus and the subsequent outcomes over time.
+
+image::difficulty_cost_line.png[Difficulty Cost Line,title="Comparison of Difficulty/Cost and Ease of Ease-of-Use/Returns Over Time in Adopting Quarkus Features"]
+
+On the _"Y-Axis"_, we define difficulty or cost in terms of story-points per sprint, reflecting the relative effort required for the features' implementation.
+This also represents costs in terms of time and resources spent in the adoption of Quarkus features. Simultaneously, ease-of-use/returns take into account metrics such as decreased debugging time,
+faster feature development, and improvements in team productivity post successful implementation.
+The graph clearly demonstrates that at the outset (tagged as _"Begin"_ on the _"X-Axis"_), both the difficulty (illustrated in higher story points) and costs are at their peak, signifying a challenging initial phase.
+However, as we move along the timeline from _"Begin"_ through _"Middle"_ and onto _"Future"_, we see a notable drop in story-points per sprint, indicating a reduced difficulty level and cost.
+In parallel to this, the ease-of-use and returns charted start at a comparatively low point at the beginning.
+These escalate gradually as we advance along the timeline towards _"Middle"_ and _"Future"_, showing a tangible increase in productivity and other gains from adopting and integrating Quarkus features into our practices.
+
+By the time we reach _"Future"_, we see a substantial decrease in difficulty and cost, while the ease-of-use and returns have considerably increased.
+This dual progression effectively highlights the significant benefits of investing in the adoption of Quarkus, despite the initial challenges.
```

Comment:
Was the initial challenge because Quarkus is hard to grasp or because you were coming from Spring Boot/Django and so were not in known territories as you needed to learn Microprofile/Jakarta EE/...?

Because I wouldn't expect the switch to Quarkus being hard if you're relatively familiar with the technology?

Anyway, maybe worth a mention if you think it makes sense.

Reply:
Hi @gsmet  

Firstly, I appreciate your comments.

I'd like to share an insight based on feedback from my colleagues, suggesting that an in-depth understanding of the standards is needed, specifically around native code execution. This necessitates a comprehensive understanding of native compilation principles, as well as a more detailed understanding of how Quarkus and Graalvm operate.

Additionally, we've developed a robust set of custom extensions. For this, I found it necessary to provide training on creating such extensions. However, I agree with the perspectives conveyed in the graphic. It's indicative of the fact that acquiring a thorough command over Quarkus's capabilities does necessitate a significant personal commitment.

Thank you.